flink-tests/src/test/java/org/apache/flink/test/recovery/BatchFineGrainedRecoveryITCase.java

```diff
@@ -81,13 +106,19 @@ public void setup() throws Exception {
 
 		miniCluster = new TestingMiniCluster(
 			new Builder()
-				.setNumTaskManagers(MAP_NUMBER)
+				.setNumTaskManagers(1)
```

Comment:
won't this always result in a full restart?

Reply:
Yes, it will. The idea is to test the backtracking with failing the finished producers. The test with loosing only some partitions will be more complicated but more general, of course. We could do it as a followup. Although I am not sure about its value atm.